src/hubverse_infrastructure/hubs/iam.py

```diff
@@ -93,20 +94,55 @@ def create_bucket_write_policy(hub_name: str):
     return bucket_write_policy
 
 
-def attach_bucket_write_policy(hub_name: str, github_role, bucket_write_policy):
+def attach_bucket_write_policy(resource_name: str, role: aws.iam.Role, bucket_write_policy: aws.iam.Policy):
     """Attach the S3 write policy to the role that Github Actions assumes."""
 
     # Update the role we created for Github Actions by attaching the
     # policy that allows writes to the hub's S3 bucket
-    aws.iam.RolePolicyAttachment(resource_name=hub_name, role=github_role.name, policy_arn=bucket_write_policy.id)
+    aws.iam.RolePolicyAttachment(resource_name=resource_name, role=role.name, policy_arn=bucket_write_policy.id)
+
+
+def create_model_output_lambda_trigger(
+    hub_name: str, hub_bucket: aws.s3.Bucket, model_output_lambda: aws.lambda_.Function
+) -> aws.s3.BucketNotification:
+    """Create the trigger that will invoke the model output lambda when a new file is written to the hub's S3 bucket."""
+    allow_bucket = aws.lambda_.Permission(
+        resource_name=f"{hub_name}-allow",
+        statement_id="AllowExecutionFromS3Bucket",
+        action="lambda:InvokeFunction",
+        function=model_output_lambda.arn.apply(lambda arn: f"{arn}"),
+        principal="s3.amazonaws.com",
+        source_arn=hub_bucket.arn.apply(lambda arn: f"{arn}"),
+    )
+
+    bucket_notification = aws.s3.BucketNotification(
+        resource_name=f"{hub_name}-create-notification",
+        bucket=hub_bucket.id,
+        lambda_functions=[
+            aws.s3.BucketNotificationLambdaFunctionArgs(
+                lambda_function_arn=model_output_lambda.arn.apply(lambda arn: f"{arn}"),
```

Comment:
This tortured-looking code: `model_output_lambda.arn.apply(lambda arn: f"{arn}")`

Is simply getting the amazon resource name (ARN) of our model-output lambda function. 

`.apply` is the Pulumi construct for awaiting the availability of resource data that doesn't exist until physically created on AWS.
(for example, if the lambda is still being created, `.apply` tells Pulumi to wait here until the lambda exists and has an ARN.)

These outputs can't be directly converted to strings, hence the Python lambda.

More info: https://leebriggs.co.uk/blog/2021/05/09/pulumi-apply